tools/proofers/dp_proof.js

```diff
@@ -230,3 +230,17 @@ function transformText(transformType) {
     txtArea.focus();
 }
 
+// Avoid submitting twice if user double-clicks button
+// Usage: onClick="return submitJustOnce(this);"
+//
+// Uses a data attribute to store when button has been clicked for the first time
+// Returns true on first click to continue submission
+// If attribute is already set (second click), return false to avoid re-submitting
+function submitJustOnce(btn) {
```

Comment:
Can we add a unit test for this method easily?

Reply:
I'm sorry, but I know nothing about that. Happy to do something if someone is able to explain what's needed

Reply:
Essentially you need to add a file here:
https://github.com/DistributedProofreaders/dproofreaders/tree/master/SETUP/tests/jsTests
Then setup a file as described by
https://qunitjs.com/
You can open https://github.com/DistributedProofreaders/dproofreaders/blob/master/SETUP/tests/qunit.html to run the test. I imagine you'll need to create an <input element programmatically for the test.

Reply:
I'm afraid I have no idea at all about any of that - if you had time, would you be willing to guide me through the process at some point, maybe under a different PR @chrismiceli ? Thanks.

Reply:
Let me find time tomorrow to see what is going on with the disable option and maybe I can send a commit with the unit test change too if that is ok?

Reply:
Thanks - that sounds like a good plan